### PR TITLE
Properly reset dask.config scheduler and shuffle when closing the client

### PR DESCRIPTION
**Pull request by gsakkis**
Fixes https://github.com/dask/distributed/issues/2473

### CODE REVIEW

File: distributed/client.py [hunk 1]
```diff
@@ -1074,9 +1071,9 @@ def _close(self, fast=False):
...
             self._scheduler_identity = {}
             with ignoring(AttributeError):
-                dask.config.set(scheduler=self._previous_scheduler)
-            with ignoring(AttributeError):
-                dask.config.set(shuffle=self._previous_shuffle)
+                # clear the dask.config set keys
+                with self._set_config:
+                    pass
             if self.get == dask.config.get('get', None):
                 del dask.config.config['get']
```
Comment (mrocklin) [inline distributed/client.py]: What if the user had specified something prior to starting the scheduler, like 

```python
dask.config.set(scheduler='threads')
with Client():
    pass
assert dask.config.get('scheduler') == 'threads'
```

Comment (mrocklin) [reply distributed/client.py]: Same comment with shuffle.

Comment (gsakkis) [reply distributed/client.py]: Just tested it, the assertion passes; the `with self._set_config` restores the previous settings at exit.

Comment (mrocklin) [reply distributed/client.py]: OK, you're right.  That's a nice solution.

**Comment by mrocklin** [comment]
This looks good to me.  It looks like there is a test failure though.  You might have to slightly modify that test.

**Comment by gsakkis** [comment]
I fixed the related test, the others that fail randomly are probably irrelevant (`test_io_loop_periodic_callbacks` on python 3.5 at one run, `test_log_tasks_during_restart` on 3.7 at another, with no code diff between them).

**Comment by mrocklin** [comment]
Thanks @gsakkis !

Merging.

**Comment by mrocklin** [comment]
Also, I notice that this is your first code contribution to this repository.  Welcome!

**Pull request merged**